### PR TITLE
Set editor mode on initialization to prevent initial text editor focus

### LINKED ISSUE #35697: [Components] Rich text editor receives focus on page load
When initially loading a page with the new `RichTextEditor` component, the editor will receive focus immediately.  Ideally, focus should not happen until a user clicks into the editor.

This is happening because we select the first block in the editor in order to show the toolbar.  Gutenberg then automatically focuses the selected block.

This fix may require a change upstream in Gutenberg in order to prevent selection from happening.

**Acceptance criteria**

* [ ] The editor does not initially receive focus when loading a product page
* [ ] The rich text editor toolbar continues to be shown prior to clicking on the editor

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR is a workaround to prevent Gutenberg's focus of initially selected elements.  Note that we need to select an element initially in order to get the toolbar to show, but we don't the focus to immediately go to these editors.

Closes #35697  .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Load the new product management experience ("Products -> Add new (MVP)")
2. On page load, make sure that the rich text editor toolbars are shown
3. Make sure that neither of the editors are focused
4. Make some changes inside an editor and make sure things continue to work as expected

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by mdperez86 — COMMENTED** [review]
Some linter problems

**Comment by joshuatf** [comment]
Thanks for noting that, @mdperez86!

Updated in https://github.com/woocommerce/woocommerce/pull/35701/commits/b186c77cbc61572d2a6e607d6faadf80226e0b1f to check for this method since the unit tests don't have access to the same block editor store.

File: packages/js/components/src/rich-text-editor/editor-writing-flow.tsx [hunk 1]
```diff
@@ -31,21 +31,38 @@ export const EditorWritingFlow = ( {
...
+	// This is a workaround to prevent focusing the block on intialization.
+	// Changing to a mode other than "edit" ensures that no initial position
+	// is found and no element gets subsequently focused.
+	// See https://github.com/WordPress/gutenberg/blob/411b6eee8376e31bf9db4c15c92a80524ae38e9b/packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js#L42
+	const setEditorIsInitializing = ( val: boolean ) => {
+		if ( typeof __unstableSetEditorMode !== 'function' ) {
+			return;
+		}
+
+		__unstableSetEditorMode( val ? 'initialized' : 'edit' );
```
Comment (nathanss) [inline packages/js/components/src/rich-text-editor/editor-writing-flow.tsx]: I know that TypeScript says explicitly it's a boolean, but reading this part with `val` made me confused for a little bit. What do you think about renaming to `isInitializing`?

**Review by mdperez86 — DISMISSED** [review]
LGTM

**Review by nathanss — APPROVED** [review]

**Pull request merged**